cmd/templ/generatecmd/main_test.go

```diff
@@ -42,3 +43,75 @@ func TestGenerate(t *testing.T) {
 		}
 	})
 }
+
+func TestDefaultWatchPattern(t *testing.T) {
+	tests := []struct {
+		name    string
+		input   string
+		matches bool
+	}{
+		{
+			name:    "empty file names do not match",
+			input:   "",
+			matches: false,
+		},
+		{
+			name:    "*_templ.txt matches, Windows",
+			input:   `C:\Users\adrian\github.com\a-h\templ\cmd\templ\testproject\strings_templ.txt`,
+			matches: true,
+		},
+		{
+			name:    "*_templ.txt matches, Unix",
+			input:   "/Users/adrian/github.com/a-h/templ/cmd/templ/testproject/strings_templ.txt",
+			matches: true,
+		},
+		{
+			name:    "*.templ files match, Windows",
+			input:   `C:\Users\adrian\github.com\a-h\templ\cmd\templ\testproject\templates.templ`,
+			matches: true,
+		},
+		{
+			name:    "*.templ files match, Unix",
+			input:   "/Users/adrian/github.com/a-h/templ/cmd/templ/testproject/templates.templ",
+			matches: true,
+		},
+		{
+			name:    "*_templ.go files match, Windows",
+			input:   `C:\Users\adrian\github.com\a-h\templ\cmd\templ\testproject\templates_templ.go`,
+			matches: true,
+		},
+		{
+			name:    "*_templ.go files match, Unix",
+			input:   "/Users/adrian/github.com/a-h/templ/cmd/templ/testproject/templates_templ.go",
+			matches: true,
+		},
+		{
+			name:    "*.go files match, Windows",
+			input:   `C:\Users\adrian\github.com\a-h\templ\cmd\templ\testproject\templates.go`,
+			matches: true,
+		},
+		{
+			name:    "*.go files match, Unix",
+			input:   "/Users/adrian/github.com/a-h/templ/cmd/templ/testproject/templates.go",
+			matches: true,
+		},
+		{
+			name:    "*.css files do not match",
+			input:   "/Users/adrian/github.com/a-h/templ/cmd/templ/testproject/templates.css",
+			matches: false,
+		},
+	}
+	wpRegexp, err := regexp.Compile(defaultWatchPattern)
+	if err != nil {
+		t.Fatalf("failed to compile default watch pattern: %v", err)
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			test := test
```

Comment:
You shouldn't need this `test := test` anymore. Old habits die hard!

Reply:
I still get a linter warning. I think it's because the `go.mod` sets the Go version at 1.21.

<img width="697" alt="Screenshot 2024-12-30 at 13 23 54" src="https://github.com/user-attachments/assets/e4d4d35f-b58a-4283-a21c-f070aff7a6cd" />

However, it's OK for us to bump up to 1.22 now, as per https://endoflife.date/go

Reply:
I'll do that in a separate change though, so it hits the release notes.

Reply:
Cool, all good then! I think once we bumb `go vet` will point out where we are doing it for no reason.